include/SugarEmailAddress/SugarEmailAddress.php

```diff
@@ -103,7 +103,7 @@ public function handleLegacySave($bean)
             || !isset($_REQUEST[$bean->module_dir . '_email_widget_id'])
             || !isset($_REQUEST['massupdate'])
         ) {
-            if (empty($this->addresses)) {
+            if (empty($this->addresses) || (!empty($bean->email1))) {
```

Comment:
I'm wondering if there could go something wrong here, eg what will happen if a bean has addresses and email1 is set? 

It will clear all the addresses and create new from the legacy address data, which I think is not as complete.

Not 100% but I think this line could cause data loss?

Reply:
Seems to be a regression issue - Dillon do you have a commit of what was changed in this file?

Reply:
@PedroErnst you were right here, this is causing issue https://github.com/salesagility/SuiteCRM/issues/7842

@mattlorimer, this needs to be reverted and fixed in another place